runtime/src/jsMain/kotlin/pbandk/internal/binary/ProtobufjsBinaryWireMarshaller.kt

```diff
@@ -0,0 +1,133 @@
+package pbandk.internal.binary
+
+import pbandk.*
+import pbandk.protobufjs.Writer
+import pbandk.protobufjs.protobufjsLong
+
+private fun Writer.writeValueNoTag(type: FieldDescriptor.Type, value: Any) {
```

Comment:
Is this code unable to use the `JsonValueMarshaller` and related classes from `commonMain`?  I guess I'm unclear about the relation of this class (and `CodedStreamBinaryWireMarshaller` below) to the ones in `commonMain`.

Reply:
I think you're confusing the JSON marshallers and the "binary" marshallers. `ProtobufjsBinaryWireMarshaller` and `CodedStreamBinaryWireMarshaller` are both responsible for converting a field into protobuf binary format. Whereas `JsonValueMarshaller` is responsible for converting a field into protobuf JSON format.

Reply:
Yes, sorry, I understand the different between the JSON and the Binary - bad example specifying `JsonValueMarshaller` in my comment.

The point I was trying to make was that it seemed like there was a lot of duplicate work being done in this `jsMain` module, that is already done in `commonMain`, but looking closer, this probably isn't the case.  I guess I just saw enough `when` statements that rolled through each type, either reading or writing the value, and I thought we wouldn't need to do that outside of `commonMain`.  But if I understand correctly, this code is just providing the javascript bindings of doing the actual reading/writing, and code up in `commonMain` is using this marshaller for doing the work in javascript.  Is my understanding correct?